llvm/lib/Target/Xtensa/XtensaInstrInfo.td

```diff
@@ -238,6 +239,11 @@ def L32R : RI16_Inst<0x01, (outs AR:$t), (ins L32Rtarget:$label),
   let imm16 = label;
 }
 
+//extending loads
```

Comment:
(not related to this PR)
I guess I'm late but still. It might make sense to split instruction definitions and patterns. E.g. extract pattern for i32 load from L32I definition and put it here together with other patterns for loads.
This will make instruction definitions more readable and patterns grouped so that you don't have to look through the entire file when searching for them.

Reply:
@s-barannikov , I prepared second commit with call lowering. Could you please take a look? We assume in the call lowering implementation that in most cases we should generate indirect call instruction and thus we need to load address constant from ConstantPool. So, I added base implementation of the ConstantPool to second commit. If it is not very complex to review both commits, I can merge them or drop second commit in other case.

Reply:
It is very difficult indeed. I left some comments and will revisit once they are addressed.